DOC: Add links to explicit vs implicit API everywhere "OO" is used (#22613)
matplotlib/matplotlib: doc/api/index.rst

Usage patterns
--------------

> Below we describe several common approaches to plotting with Matplotlib.  See

Reviewer: ```suggestion
Below we describe several common approaches to plotting with Matplotlib. See
```

> :ref:`api_interfaces` for an explanation of the trade off of the supported user

Reviewer: ```suggestion
:ref:`api_interfaces` for an explanation of the trade-offs between the supported user
```

APIs.

> The Explicit API

Reviewer: ```suggestion
The explicit API
```

^^^^^^^^^^^^^^^^

At its core, Matplotlib is an object-oriented library. We recommend directly

> working with the objects, if you need more control and customization of your

Reviewer: ```suggestion
working with the objects if you need more control and customization of your
```

plots.

In many cases you will create a `.Figure` and one or more
`~matplotlib.axes.Axes` using `.pyplot.subplots` and from then on only work
on these objects. However, it's also possible to create `.Figure`\ s
explicitly (e.g. when including them in GUI applications).

Further reading:

- `matplotlib.axes.Axes` and `matplotlib.figure.Figure` for an overview of
   plotting functions.
- Most of the :ref:`examples <examples-index>` use the object-oriented approach

>    (except for the pyplot section)

Reviewer: This renders a bit weird with bold on the first line and standard text on the second for these bullets.
